packages/mongodb-ts-autocomplete/scripts/extract-types.ts

```diff
@@ -1,43 +1,257 @@
+    // There's a chicken-and-egg problem here: we can't compile this module
+    // itself without the extracted types and we can't extract it before it is
+    // compiled because the module is not compiled, so its dist/index.js does
+    // not exist yet. With this workaround we're resolving the package.json
+    // which definitely already exists.
+    const result = {
+      resolvedModule: {
+        resolvedFileName: path.resolve(__dirname, '..', 'package.json'),
+        packageId: {
+          subModuleName: 'package.json',
+        },
+      },
+    };
+
+    return result;
+  }
+  const result = ts.resolveModuleName(
+    moduleName,
+    __filename,
+    {},
+    {
+      fileExists: (path: string) => ts.sys.fileExists(path),
+      readFile: (path: string) => ts.sys.readFile(path),
+    },
+  );
+
+  return result;
+}
+
+function resolveModulePath(moduleName: string): string {
+  const { resolvedModule } = resolve(moduleName);
+  if (!resolvedModule) {
+    throw new Error(`Could not resolve module: ${moduleName}`);
+  }
+  return resolvedModule.resolvedFileName;
+}
+
+const deps: Record<string, string[]> = {
+  '@mongodb-js/mongodb-ts-autocomplete': [
+    // the module resolution won't be addressing this module by name, but we'll
+    // feed it this package.json as a fallback when it tries to find itself
+    'package.json',
+  ],
+  '@types/node': [
+    'package.json',
+    'assert.d.ts',
+    'assert/strict.d.ts',
+    'async_hooks.d.ts',
+    'buffer.buffer.d.ts',
+    'buffer.d.ts',
+    'child_process.d.ts',
+    'cluster.d.ts',
+    'compatibility/disposable.d.ts',
+    'compatibility/index.d.ts',
+    'compatibility/indexable.d.ts',
+    'compatibility/iterators.d.ts',
+    'console.d.ts',
+    'constants.d.ts',
+    'crypto.d.ts',
+    'dgram.d.ts',
+    'diagnostics_channel.d.ts',
+    'dns.d.ts',
+    'dns/promises.d.ts',
+    'dom-events.d.ts',
+    'domain.d.ts',
+    'events.d.ts',
+    'fs.d.ts',
+    'fs/promises.d.ts',
+    'globals.d.ts',
+    'globals.typedarray.d.ts',
+    'http.d.ts',
+    'http2.d.ts',
+    'https.d.ts',
+    'index.d.ts',
+    'inspector.d.ts',
+    'module.d.ts',
+    'net.d.ts',
+    'os.d.ts',
+    'path.d.ts',
+    'perf_hooks.d.ts',
+    'process.d.ts',
+    'punycode.d.ts',
+    'querystring.d.ts',
+    'readline.d.ts',
+    'readline/promises.d.ts',
+    'repl.d.ts',
+    'sea.d.ts',
+    'sqlite.d.ts',
+    'stream.d.ts',
+    'stream/consumers.d.ts',
+    'stream/promises.d.ts',
+    'stream/web.d.ts',
+    'string_decoder.d.ts',
+    'test.d.ts',
+    'timers.d.ts',
+    'timers/promises.d.ts',
+    'tls.d.ts',
+    'trace_events.d.ts',
+    'tty.d.ts',
+    'url.d.ts',
+    'util.d.ts',
+    'v8.d.ts',
+    'vm.d.ts',
+    'wasi.d.ts',
+    'worker_threads.d.ts',
+    'zlib.d.ts',
+  ],
+  assert: [
+    'package.json',
+    'assert.js', // exists only
+  ],
+  buffer: ['package.json', 'index.d.ts'],
+  events: ['package.json'],
```

Comment:
Not sure how important this is, probably not much, but I think these should be resolving to node built-ins and not to the npm packaged reimplementations of those, just wondering why they are getting here